build.yaml

```diff
@@ -196,6 +198,15 @@ common_cli:
     # instance running on the Mesos Slave where the job is running.
     export eth0_ip=\$( ip -o -4 addr show eth0 |awk '{ print \$4 '} | cut -f 1 -d "/")
 
+    # pass the exit code from the previous and return the the aggregated
+    # exit status for the whole job.
+    function updateExitStatus {
+      if [ "\${1}" !=  "0" ]; then
+        echo 1
+      fi
+    }
+
+
   # TODO: do we need to clean up old files on ubuntu and centos or
```

Comment:
we have a large number of actions that can fail, and we need to make we go through the job and don't abort it straight away as we need to do some cleanup actions.
we use the updateExitStatus() to manipulate the global exit code of the job